PySCF.py

```diff
@@ -80,6 +80,33 @@ def is_consistent(self,other):
             .format(key,self.__dict__[key],other.__dict__[key]))
         return False
     return True
+
+  #---------------------------------------------------
+  def update_input(self, fname):
+    # check if input file already updated
+    lines = open(fname, 'r').read().split()
```

Comment:
Here we're editing a file. This is a rather fragile thing to do. I would prefer to rewrite the file.